src/Microsoft.AspNetCore.Razor.Evolution/RazorSourceDocument.cs

```diff
@@ -17,6 +17,8 @@ public abstract class RazorSourceDocument
 
         public abstract int Length { get; }
 
+        public abstract RazorSourceLineCollection Lines { get; }
```

Comment:
We already expose it as a collection of characters though. I feel like it'd be overkill to add yet another abstraction on how to read the same document on the abstract base type. Maybe the approach of `SourceDocumentReader` that linified the document and enabled retrieval of source locations?